### ISSUE TITLE
Support cron builds

### ISSUE THREAD

**Issue by simonbrandhof** [cirruslabs/cirrus-ci-docs#39]
Cron builds would allow to schedule the execution of a build on a specified branch, at fixed intervals, independently of whether any commits were pushed to the repository. 
Configuring a cron build from cirrus-ci.com would require to define the interval (hourly, daily, weekly or a custom value) and the branch (by default the branch marked as "default" in GitHub should be used).

Cron builds send notifications as any other builds.

**Comment by cevich** [comment]
This also supports having a polling 'bot' which could then do maintenance tasks on a repository.  For example, checking all open PRs for a 'new' keyword like: '/bot: rebuild images' :wink:

**Comment by dbolkensteyn** [comment]
Any update on this @fkorotkov ?

I'd also be interested in programmatically starting a build using the GraphQL API, and essentially use Cirrus CI as a layer of abstraction over the various backends it supports (GCE, GKE, Azure, MacStadium, etc.) - which is slightly related to this use-case.

**Comment by simonbrandhof** [comment]
For reference: #90

**Comment by gspencergoog** [comment]
Hey @fkorotkov, any update on this feature?

**Comment by RDIL** [comment]
Any feature requests that do not have the `coming-soon` label are on hold.

**Comment by fkorotkov** [comment]
This feature will come after #90 and #74. This feature is not that trivial to make is scalable (GCP even has [it's own paid version](https://cloud.google.com/scheduler/))

**Comment by gspencergoog** [comment]
Thanks for the update.  I never thought it was trivial. :-)

**Comment by fkorotkov** [comment]
Initially I though it is pretty simple but the more thoughts I put into it the more complicated it appears. 😅

**Comment by simonbrandhof** [comment]
@fkorotkov I don't know at all your constraints, so just FYI: https://cloud.google.com/scheduler/ is an interesting service from GCP.

**Comment by duckinator** [comment]
Thank you so much for adding this!
